backend/core/src/primary-adapters/rest/user/user-controller.ts

```diff
@@ -58,19 +79,101 @@ export class UserController extends BaseHttpController {
    *        401:
    *          $ref: '#/components/responses/Unauthorized'
    */
-  @httpGet('/', authenticationMiddleware)
+  @httpGet(UserApiPath.ROOT, authenticationMiddleware)
   public getAllUsers(): Promise<User[]> {
     return this.userService.getAllUsers();
   }
 
-  //NOTE: this routes only for testing and in future should removed or modified
-  @httpPost('/notify')
-  public notify(@requestBody() body: { data: { message: string } }): Promise<boolean> {
-    return this.userService.notify(body);
+  /**
+   * @swagger
+   * /users/bind/{id}:
+   *    post:
+   *        tags:
+   *          - users
+   *        summary: Bind user preferences
+   *        security:
+   *          - bearerAuth: []
+   *        parameters:
+   *          - name: id
+   *            required: true
+   *            in: path
+   *            schema:
+   *              type: string
+   *        requestBody:
+   *          description: Categories
+   *          required: true
+   *          content:
+   *            application/json:
+   *              schema:
+   *                type: object
+   *                properties:
+   *                  categories:
+   *                    type: array
+   *                    items:
+   *                      type: string
+   */
+  @httpPost(UserApiPath.$BIND, authenticationMiddleware)
+  public bindCategories(
+    @requestBody() payload: Omit<UserBindCategoriesDto, 'id'>,
+    @requestParam() { id }: DefaultRequestParam,
+    @request() req: ExtendedAuthenticatedRequest,
+  ): Promise<CategoryResponseDto[]> {
+    //should create middleware?
+    const { id: clientUserId } = req.user;
```

Comment:
why do you even pass client id in request dto and throw error in case of mismatch, if the user's id is available on request without further info? just use req.user.id without any comparisons and excessive request payload